mesonbuild/dependencies/dev.py

```diff
@@ -568,6 +568,17 @@ def __init__(self, environment: 'Environment', kwargs: JNISystemDependencyKW):
 
         m = self.env.machines[self.for_machine]
 
+        if m.is_android():
+            if kwargs.get('version'):
+                mlog.warning('Ignoring JNI version requirements as there is no mechanism to determine the version')
```

Comment:
s/as/because for grammar purposes. Maybe we should also suffix this with `on Android`